internal/backends/sqlite/metadata/registry.go

```diff
@@ -537,6 +537,11 @@ func (r *Registry) indexesDrop(ctx context.Context, dbName, collectionName strin
 		c.Settings.Indexes = slices.Delete(c.Settings.Indexes, i, i+1)
 	}
 
+	q := fmt.Sprintf("UPDATE %q SET settings = ?", metadataTableName)
+	if _, err := db.ExecContext(ctx, q, c.Settings); err != nil {
```

Comment:
If this fails, do we want have undo `drop indexes`? I see that `indexesCreate(..)` attempts undo action :thinking: . After all undo action is also best effort though.

Reply:
Then we might be in an infinite loop because `indexesCreate` calls `dropIndexes` if a similar thing fails.
A better approach would be to use a transaction but we decided against transactions.

Reply:
Yeah if we call `indexesCreate` from this, it will produce infinite loop. So solution would need to be more than just that and it would bring more logic. So for now, let's keep it as it is :hugs:

Reply:
Yeah... I'm a bit worried though, as with the current approach we don't have enough consistency. So, maybe having a potential infinite loop is better than having a potential data inconsistency...

@noisersup @AlekSi what do you think?